### PR TITLE
FeatureFinderMetabo: param to toggle reporting smoothed intensities

### PR DESCRIPTION
**Pull request by rickhelmus**
## Description

This small PR fixes issue #7582 by adding an extra param to FeatureFinderMetabo (`report_smoothed_intensities`) that controls whether smoothed intensities are reported or not.

## Checklist
- [X] Make sure that you are listed in the AUTHORS file
- [x] Add relevant changes and new features to the CHANGELOG file
- [X] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

___

### **PR Type**
enhancement

___

### **Description**
- Introduced a new parameter `report_smoothed_intensities` in `FeatureFindingMetabo` to control whether smoothed intensities are reported.
- Updated the logic in `FeatureFindingMetabo.cpp` to utilize the new parameter for intensity calculations.
- Added a new boolean member `report_smoothed_intensities_` in the header file to support the new functionality.
- Updated the AUTHORS file to include Rick Helmus as a contributor.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>FeatureFindingMetabo.cpp</strong><dd><code>Add parameter to toggle reporting of smoothed intensities</code></dd></summary>
<hr>

src/openms/source/FEATUREFINDER/FeatureFindingMetabo.cpp

<li>Added a new parameter <code>report_smoothed_intensities</code> to control reporting <br>of smoothed intensities.<br> <li> Updated logic to use <code>report_smoothed_intensities</code> for intensity <br>calculations.<br>

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7594/files#diff-c4295d51055a96842e571b1d932ddb5bd2651241d39ff9eee20c7c7fe081a1c7">+7/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>FeatureFindingMetabo.h</strong><dd><code>Declare new member for smoothed intensities reporting</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/openms/include/OpenMS/FEATUREFINDER/FeatureFindingMetabo.h

- Introduced a new boolean member `report_smoothed_intensities_`.

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7594/files#diff-77b34be844d3ffbfd0a62300a12f2539c258ea9f46f10b7e9c945d0ef4212946">+2/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>AUTHORS</strong><dd><code>Update AUTHORS file with new contributor</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

AUTHORS

- Added Rick Helmus to the authors list.

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7594/files#diff-ab6af77435f58cc0c9d4c31dfe05656e45187cc7c7fc02aada401a7642125463">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

**Comment by rickhelmus** [comment]
Hi all, this is my first PR here, so I could use some hints to tick the checkboxes still open:

* Where to put the changes in the `CHANGELOG`? Since 3.2 was just released, I guess some new bullets are needed?
* I am unsure whether this minor change warrents any unit tests and/or Python changes

Thanks,
Rick

File: src/openms/source/FEATUREFINDER/FeatureFindingMetabo.cpp [hunk 1]
```diff
@@ -307,6 +309,7 @@ namespace OpenMS
     
     isotope_filtering_model_ = param_.getValue("isotope_filtering_model").toString();
     use_smoothed_intensities_ = param_.getValue("use_smoothed_intensities").toBool();
+    report_smoothed_intensities_ = use_smoothed_intensities_ && param_.getValue("report_smoothed_intensities").toBool();
```
Comment (timosachsenberg) [inline src/openms/source/FEATUREFINDER/FeatureFindingMetabo.cpp]: we could issue an OPENMS_LOG_WARN if "report_smoothed_intensities" is true but "use_smoothed_intensities" is false

**Comment by timosachsenberg** [comment]
> * Where to put the changes in the `CHANGELOG`? Since 3.2 was just released, I guess some new bullets are needed?

Adding it under library is sufficient. We can also clean it up if needed.
 https://github.com/OpenMS/OpenMS/blob/develop/CHANGELOG#L31

> * I am unsure whether this minor change warrents any unit tests and/or Python changes
I think tests are not super important here but there is one thing I would love to see.
How should we document this in the pyOpenMS pxd file and the C++ header file such that:
- users learn about the new option (easy ;) )
- when enabling/disabling the option is appropriate. In particular, I don't fully understand the impact of disabling smoothing outside of reporting the values (e.g., does this affect in any way elution boundary detection etc.)

If you have some insights here it would be highly appreciated if you could add some information for other users.

**Comment by rickhelmus** [comment]
Hi @timosachsenberg ,

Thanks for your feedback and commits!

For the documentation: my main purpose was to export the 'original' intensities, so the user can more easily link the reported features to the raw HRMS data in e.g. chromatograms and other plots. Then by keeping smoothing enabled I guess feature detection can be improved for eg noisy peaks.

PS: coincidently I found that `MRMTransitionGroupPicker` has a similar parameter (`peak_integration`).
PPS: I intended to rebase instead of adding a merge commit, if you prefer me to e.g. squash please let me know.

**Comment by timosachsenberg** [comment]
Great! You can also squash commits if you want. No real preference.

**Pull request merged**